src/bank02.asm

```diff
@@ -8130,29 +8126,197 @@ titleScreenLicenseText:
-    TXT  "<00>"                                        ;; 02:7f2b ?
-    TXT  "Legend tells that<00>"                       ;; 02:7f2c ??????????????????
-    TXT  "it gives eternal<00>"                        ;; 02:7f3e ?????????????????
-    TXT  "power to the one<00>"                        ;; 02:7f4f ?????????????????
-    TXT  " who touched it.<00>"                        ;; 02:7f60 ?????????????????
-    TXT  "<00>"                                        ;; 02:7f71 ?
-    TXT  "  Dark Lord was<00>"                         ;; 02:7f72 ????????????????
-    TXT  "  trying to find<00>"                        ;; 02:7f82 ????????????????.
-    TXT  "  the way to the<00>"                        ;; 02:7f93 ...........??????
-    TXT  " Tree of Mana to<00>"                        ;; 02:7fa4 ?????????????????
-    TXT  "  get the mighty<00>"                        ;; 02:7fb5 ?????????????????
-    TXT  " power to conquer<00>"                       ;; 02:7fc6 ??????????????????
-    TXT  "    the world.<00>"                          ;; 02:7fd8 ???????????????
-    TXT  "<00>"                                        ;; 02:7fe7 ?
-    db   $00, $00, $00, $00, $00, $00, $00, $00        ;; 02:7fe8 ????????
-    db   $00, $00, $00, $00, $01, $ff, $ff, $ff        ;; 02:7ff0 ????????
-    db   $ff, $ff, $ff, $ff, $ff, $ff, $ff, $ff        ;; 02:7ff8 ????????
+    TXT  "<00>"                                        ;; 02:7fd8 ?
+    db   $01                                           ;; 02:7fd9 ?
+
+; Draws left aligned number HL as modified tiles at WRAM position DE
+; C holds the drawing mode in the first nibble (0 for shift, 1 for shift/swap)
+; C holds number of max digits in the second nibble
+drawLeftAlignedNumberInWRAM:
+    ld   B, $00 ; digit counter
+
+    ; Housekeeping, get a copy of proper WRAM starting
+    ;  location on the stack for later use
+    push BC
+    push DE
+
+    ; Load up the digits on the stack
+.divmod_and_store_number:
+    ld   A, $0a
+    push BC
+    push DE
+    call divMod
+    pop  DE
+    pop  BC
+    push AF
+    inc  B
+    ld   A, H
+    or   A, L
+    jr   NZ, .divmod_and_store_number
+
+    ; Reduce second nibble of C by num digits found
+    ; Remainder provides the number of tiles to clear
+    ; after we write the digits
+    ld   A, C
+    sub  A, B
+    ld   C, A
+
+    ; Increment tile buffer pointer (DE) if in swap mode
+    ; since we will be swapping back 4px to the tile before
+    ld   H, D
+    ld   L, E
+    ld   D, $00
+    and  A, $10
+    ld   E, A
+    add  HL, DE
+    ld   D, H
+    ld   E, L
+
+    ; Stack holds B digits to pop, transfer them
+ .grab_digit_and_transfer:
+    ld   HL, gfxStatusBar+$100 ; start of number tiles
+    pop  AF ; grab digit
+    swap A
+    push DE
+    ld   D, $00
+    ld   E, A
+    add  HL, DE
+    pop  DE
+    ld   A, BANK(gfxStatusBar)
+    push BC
+    ld   B, $10
+    call copyAndRotateBankAfromHLtoDE
```

Comment:
Is this a candidate for inlining? Do you foresee wanting to use it additional places in the future?

Reply:
I think it needs to be a function in bank 0 in order to swap to bank 8 for the copy. Please let me know if that's not the case though... maybe my bank understanding is lacking.

Reply:
Good point. I was thinking of locating all of this into a new bank, but that would mean adding a second copy of this set of numbers. Still an option, but I'll try to find space in bank 2.

Any method of preventing multiple bank switches (aside from locating the code with the tiles) I can think of would take up more bank 0 space, and right now this fits well.